pkg/store/proxy.go

```diff
@@ -371,52 +475,30 @@ func startStreamSeriesSet(
 				s.warnCh.send(storepb.NewWarnSeriesResponse(errors.New(w)))
 				continue
 			}
-
+			metrics.recvChannelSize.WithLabelValues(storeAddr).Observe(float64(len(s.recvCh)))
+			if len(s.recvCh) == 10 {
+				metrics.writesToBlockedChannel.WithLabelValues(storeAddr).Inc()
```

Comment:
it seems that `channelBlockedDuration ` will be enough and I'm not sure we need this metric at all.
Also `== 10` is a magic number here :)

Reply:
let's remove it